erigon-lib/state/domain_shared.go

```diff
@@ -856,22 +858,11 @@ func (sd *SharedDomains) DomainDelPrefix(domain kv.Domain, prefix []byte) error
 	if domain != kv.StorageDomain {
 		return fmt.Errorf("DomainDelPrefix: not supported")
 	}
-	type tuple struct {
-		k, v []byte
-		step uint64
-	}
-	tombs := make([]tuple, 0, 8)
 	if err := sd.IterateStoragePrefix(prefix, func(k, v []byte, step uint64) error {
-		tombs = append(tombs, tuple{k, v, step})
-		return nil
+		return sd.DomainDel(kv.StorageDomain, k, nil, v, step)
```

Comment:
Are you sure this is safe to mutate `SharedDomains.storage` while iterating it?

Reply:
previously we needed tombstones - no buffered io was supported. now with buffered writes and storage map is loaded into heap to sort before iterate should be good. tested yesterday along with race detector, no validity issues has been met

Reply:
Show “ buffered writes ” in code plz

Reply:
ye i was wrong about iterator loaded on heap, instead it's btree iter, sorry for this. removed this update to tombstone accumulation.